x-pack/plugins/security_solution/cypress/e2e/exceptions/entry/use_value_list.cy.ts

```diff
@@ -0,0 +1,132 @@
+import {
+  CLOSE_ALERTS_CHECKBOX,
+  EXCEPTIONS_TABLE_MODAL,
+  EXCEPTIONS_TABLE_MODAL_CONFIRM_BTN,
+  EXCEPTION_CARD_ITEM_CONDITIONS,
+  EXCEPTION_CARD_ITEM_NAME,
+  EXCEPTION_ITEM_VIEWER_CONTAINER,
+  NO_EXCEPTIONS_EXIST_PROMPT,
+} from '../../../screens/exceptions';
+
+const goToRulesAndOpenValueListModal = () => {
+  visitWithoutDateRange(DETECTIONS_RULE_MANAGEMENT_URL);
+  waitForListsIndex();
+  waitForValueListsModalToBeLoaded();
+  openValueListsModal();
+};
+
+describe('Use Value list in exception entry', () => {
+  before(() => {
+    login();
+    esArchiverLoad('exceptions');
+    deleteAlertsAndRules();
+    createRule({
+      ...getNewRule(),
+      query: 'user.name:*',
+      index: ['exceptions*'],
+      exceptions_list: [],
+      rule_id: '2',
+    });
+    visitWithoutDateRange(DETECTIONS_RULE_MANAGEMENT_URL);
+  });
+  beforeEach(() => {
+    createListsIndex();
+  });
+
+  afterEach(() => {
+    esArchiverUnload('exceptions');
+  });
+
+  it('Should use value list in exception entry, and validate deleting value list prompt', () => {
+    const ITEM_NAME = 'Exception item with value list';
+    const ITEM_FIELD = 'agent.name';
+
+    goToRulesAndOpenValueListModal();
+
+    // Add new value list of type keyword
+    const listName = 'value_list.txt';
+    selectValueListType('keyword');
+    selectValueListsFile(listName);
+    uploadValueList();
+
+    cy.get(VALUE_LISTS_TABLE)
+      .find(VALUE_LISTS_ROW)
+      .should(($row) => {
+        expect($row.text()).to.contain(listName);
+        expect($row.text()).to.contain('Keywords');
+      });
+    goToRuleDetails();
+    goToExceptionsTab();
+
+    // open add exception modal
+    openExceptionFlyoutFromEmptyViewerPrompt();
+
+    // add exception item name
+    addExceptionFlyoutItemName(ITEM_NAME);
+
+    addExceptionEntryFieldValue(ITEM_FIELD, 0);
+    addExceptionEntryOperatorValue('is in list', 0);
+
+    addExceptionEntryFieldMatchIncludedValue('value_list.txt', 0);
+
+    // The Close all alerts that match attributes in this exception option is disabled
+    cy.get(CLOSE_ALERTS_CHECKBOX).should('exist');
+    cy.get(CLOSE_ALERTS_CHECKBOX).should('have.attr', 'disabled');
+
+    // Create exception
+    submitNewExceptionItem();
+
+    // displays existing exception items
+    cy.get(EXCEPTION_ITEM_VIEWER_CONTAINER).should('have.length', 1);
+    cy.get(NO_EXCEPTIONS_EXIST_PROMPT).should('not.exist');
+    cy.get(EXCEPTION_CARD_ITEM_NAME).should('have.text', ITEM_NAME);
+    cy.get(EXCEPTION_CARD_ITEM_CONDITIONS).should(
+      'have.text',
+      ` ${ITEM_FIELD}included in value_list.txt`
+    );
+
+    // Go back to value list to delete the existing one
+    goToRulesAndOpenValueListModal();
+
+    deleteValueListsFile(listName);
+
+    // Toast should be shown because of exception reference
+    cy.get(EXCEPTIONS_TABLE_MODAL).should('exist');
+    cy.get(EXCEPTIONS_TABLE_MODAL_CONFIRM_BTN).first().click();
```

Comment:
Why are we finishing the method with an action?

Reply:
Good point, removed the extra action